itext/itext.kernel/itext/kernel/geom/IShape.cs

```diff
@@ -58,5 +58,12 @@ public interface IShape {
         /// consisting of shape's base points.
         /// </returns>
         IList<Point> GetBasePoints();
+        
+        /// <summary>
+        /// Transforms the shape by the specified matrix
+        /// </summary>
+        /// <param name="ctm">the matrix for the transformation</param>
+        /// <returns>the transformed shape</returns>
+        IShape TransformToUserspace(Matrix ctm);
```

Comment:
The Open Closed Principle(second SOLID principle) You need to extend abstractions! Will add a new interface with a IShape TransformToUserspace method!